src/systems/optical_tactile_plugin/Visualization.cc

```diff
@@ -0,0 +1,212 @@
+  _positionMarkerMsg.set_visibility(ignition::msgs::Marker::GUI);
+  ignition::msgs::Set(_positionMarkerMsg.mutable_scale(),
+    ignition::math::Vector3d(1, 1, 1));
+
+  _forceMarkerMsg.set_ns("forces_" + this->modelName);
+  _forceMarkerMsg.set_id(1);
+  _forceMarkerMsg.set_action(ignition::msgs::Marker::ADD_MODIFY);
+  _forceMarkerMsg.set_type(ignition::msgs::Marker::LINE_LIST);
+  _forceMarkerMsg.set_visibility(ignition::msgs::Marker::GUI);
+
+  // Set material properties and lifetime
+  ignition::msgs::Set(_positionMarkerMsg.mutable_material()->
+    mutable_ambient(), math::Color(0, 0, 1, 1));
+  ignition::msgs::Set(_positionMarkerMsg.mutable_material()->
+    mutable_diffuse(), math::Color(0, 0, 1, 1));
+  _positionMarkerMsg.mutable_lifetime()->set_nsec(
+    this->cameraUpdateRate * 1000000000);
+
+  ignition::msgs::Set(_forceMarkerMsg.mutable_material()->
+    mutable_ambient(), math::Color(0, 1, 0, 1));
+  ignition::msgs::Set(_forceMarkerMsg.mutable_material()->
+    mutable_diffuse(), math::Color(0, 1, 0, 1));
+  _forceMarkerMsg.mutable_lifetime()->set_sec(
+    this->cameraUpdateRate * 1000000000);
+}
+
+//////////////////////////////////////////////////
+void OpticalTactilePluginVisualization::AddNormalForceToMarkerMsgs(
+  ignition::msgs::Marker &_positionMarkerMsg,
+  ignition::msgs::Marker &_forceMarkerMsg,
+  ignition::math::Vector3f &_position,
+  ignition::math::Vector3f &_normalForce,
+  ignition::math::Pose3f &_sensorWorldPose)
+{
+  // Check if messages have been initialized
+  if (!this->normalForcesMsgsAreInitialized)
+  {
+    this->InitializeNormalForcesMarkerMsgs(_positionMarkerMsg,
+      _forceMarkerMsg);
+    this->normalForcesMsgsAreInitialized = true;
+  }
+
+  // We need to compute the two points that form a normal force (start and end
+  // points) with reference to the simulation origin
+  ignition::math::Vector3f normalForcePositionFromSensor(
+    _position.X(), _position.Y(), _position.Z());
+
+  ignition::math::Quaternionf normalForceOrientationFromSensor;
+  normalForceOrientationFromSensor.From2Axes(
+    ignition::math::Vector3f(0, 0, 1), _normalForce);
+
+  ignition::math::Pose3f normalForcePoseFromSensor(
+    normalForcePositionFromSensor, normalForceOrientationFromSensor);
+
+  ignition::math::Pose3f normalForcePoseFromWorld =
+    (normalForcePoseFromSensor + this->depthCameraOffset) + _sensorWorldPose;
+  normalForcePoseFromWorld.Correct();
+
+  // Get the first point of the normal force
+  ignition::math::Vector3f firstPointFromWorld =
+    normalForcePoseFromWorld.Pos();
+
+  // Move the normal force pose a distance of forceLength along the direction
+  // of _normalForce and get the second point
+  normalForcePoseFromSensor.Set(normalForcePositionFromSensor +
+    _normalForce * this->forceLength, normalForceOrientationFromSensor);
+
+  normalForcePoseFromWorld =
+    (normalForcePoseFromSensor + this->depthCameraOffset) + _sensorWorldPose;
```

Comment:
Consider using `Pose3f::operator*` instead of `Pose3f::operator+` since the latter is [discouraged](https://github.com/ignitionrobotics/ign-math/issues/60). If you do, note that you'll have to swap the order of operations